packetbeat/scripts/mage/build.go

```diff
@@ -17,11 +17,37 @@
 
 package mage
 
-import devtools "github.com/elastic/beats/v7/dev-tools/mage"
+import (
+	"errors"
+	"os"
+	"strings"
+
+	devtools "github.com/elastic/beats/v7/dev-tools/mage"
+)
 
 // CrossBuild cross-builds the beat for all target platforms.
 func CrossBuild() error {
-	// Run all builds serially to try to address failures that might be caused
-	// by concurrent builds. See https://github.com/elastic/beats/issues/24304.
-	return devtools.CrossBuild(devtools.Serially())
+	return devtools.CrossBuild(
+		// Run all builds serially to try to address failures that might be caused
+		// by concurrent builds. See https://github.com/elastic/beats/issues/24304.
+		devtools.Serially(),
+
+		devtools.ImageSelector(func(platform string) (string, error) {
+			image, err := devtools.CrossBuildImage(platform)
+			if err != nil {
+				return "", err
+			}
+			if os.ExpandEnv("CI") != "true" {
+				return image, nil
+			}
+			if platform == "windows/amd64" || platform == "windows/386" {
```

Comment:
I was expecting that there would be hard-coded NPCAP image tag version to ensure builds of old versions are reproducible.

Also outside of CI, if I'm signed in to the docker registry and I want to test with the npcap image I would like that to be an option somehow.

Reply:
Sure. I was thinking the NPCAP_VERSION var would be provided by the build system. I've hard coded that now, and made a second envvar to parallel the CI var so that the build can be invoked to know that the build is happening locally but the Npcap image should be used. It would be nice to automagically determine whether the image is available, but I don't see a way to do that.